sdk/src/main/java/io/radar/sdk/RadarLocationManager.kt

```diff
@@ -106,21 +95,25 @@ internal class RadarLocationManager(
             RadarTrackingOptionsDesiredAccuracy.NONE -> LocationRequest.PRIORITY_NO_POWER
         }
 
-        val locationRequest = LocationRequest().apply {
-            priority = desiredPriority
-            interval = 1000L
-            fastestInterval = 1000L
-            numUpdates = 1
-        }.setExpirationDuration(20000L)
-
         logger.d("Requesting location")
 
-        locationClient.requestLocationUpdates(locationRequest, object : LocationCallback() {
-            override fun onLocationResult(result: LocationResult?) {
-                locationClient.removeLocationUpdates(this)
-                locationManager.handleLocation(result?.lastLocation, source)
+        locationClient.getCurrentLocation(desiredPriority, null).addOnSuccessListener { location ->
+            if (location == null) {
+                logger.d("Location timeout")
+
+                callback?.onComplete(RadarStatus.ERROR_LOCATION)
+            } else {
+                logger.d("Location request success")
+
+                locationManager.handleLocation(location, source)
             }
-        }, Looper.getMainLooper())
+
+
+        }.addOnCanceledListener {
```

Comment:
I guess it's good to add this listener, though we don't supply a cancelation token